src/lib/analysis-collector-base.spec.ts

```diff
@@ -350,6 +350,10 @@ describe('AnalysisCollectorBase', () => {
       const results = await collector.spawn('echo');
       expect(results).toEqual('');
     });
+
+    it('should throw an error if the spawn command fails', async () => {
+      await expectAsync(collector.spawn('cat', ['non_existent_file'])).toBeRejected();
```

Comment:
This is effectively an integration test – we should `spyOn` this simulate an exception.

```typescript
spyOn(collector, 'spawn').and.returnValue(Promise.reject());
```

Reply:
We should also use `.toBeRejectedWith()` to ensure we're capturing the output correctly.

Reply:
It looks like this was implemented in the trivy test, but should be similar here.

Reply:
If we use a `spyOn` here, we would not have coverage on some parts of the `AnalysisCollectorBase.spawn()`

https://github.com/quantum-sec/ci-analysis-collector/blob/fix/EN-724/src/lib/analysis-collector-base.ts#L190-L193

Reply:
@GerardSetho the base class should have its own tests. You can get coverage there by doing something like:

```typescript
spyOn(service.logger, 'debug');
spyOn(proc.stderr, 'on').and.callFake((eventName, delegate) => {
  delegate('TEST_ERROR_CHUNK');
});

expect(...).toHaveBeenRejectedWith('TEST_ERROR_CHUNK');
```